consensus/ethash/sealer.go

```diff
@@ -344,8 +344,10 @@ func (s *remoteSealer) loop() {
 //   result[3], hex encoded block number
 func (s *remoteSealer) makeWork(block *types.Block) {
 	hash := s.ethash.SealHash(block.Header())
+	epochLength := calcEpochLength(block.NumberU64(), s.ethash.config.ECIP1099Block)
+	epoch := calcEpoch(block.NumberU64(), epochLength)
 	s.currentWork[0] = hash.Hex()
-	s.currentWork[1] = common.BytesToHash(SeedHash(block.NumberU64())).Hex()
+	s.currentWork[1] = common.BytesToHash(SeedHash(epoch*epochLength + 1)).Hex()
```

Comment:
`SeedHash` is also used in the `debug` API. 

```
// SeedHash retrieves the seed hash of a block.
func (api *PublicDebugAPI) SeedHash(ctx context.Context, number uint64) (string, error) {
	block, _ := api.b.BlockByNumber(ctx, rpc.BlockNumber(number))
	if block == nil {
		return "", fmt.Errorf("block #%d not found", number)
	}
	return fmt.Sprintf("0x%x", ethash.SeedHash(number)), nil
}
```

> `internal/ethapi/api.go`

Does the logic there need to change too?

Reply:
Done in subsequent commits.